tests/environment.py

```diff
@@ -0,0 +1,175 @@
+
+rootdir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+installdir = os.path.join(rootdir, 'build', 'install')
+
+
+class AlEntityDocker(AlEntity):
+    '''Docker implementation of AlEntity.
+
+    The entity is defined from the name of the container, the rest is derived from that.
+    '''
+    def __init__(self, name: str, is_controller: bool = False):
+        self.name = name
+        self.bridge_name = 'br-lan'
+
+        # First, get the UCC port from the config file
+        if is_controller:
+            config_file_name = 'beerocks_controller.conf'
+        else:
+            config_file_name = 'beerocks_agent.conf'
+        with open(os.path.join(installdir, 'config', config_file_name)) as config_file:
+            ucc_port = \
+                re.search(r'ucc_listener_port=(?P<port>[0-9]+)', config_file.read()).group('port')
+
+        device_ip_output = self.command('ip', '-f', 'inet', 'addr', 'show', self.bridge_name)
+        device_ip = re.search(r'inet (?P<ip>[0-9.]+)', device_ip_output.decode('utf-8')).group('ip')
+
+        ucc_socket = UCCSocket(device_ip, ucc_port)
+        mac = ucc_socket.dev_get_parameter('ALid')
+
+        super().__init__(mac, ucc_socket, is_controller)
+
+        # We always have two radios, wlan0 and wlan2
+        RadioDocker(self, "wlan0")
+        RadioDocker(self, "wlan2")
+
+    def command(self, *command: str) -> bytes:
+        '''Execute `command` in docker container and return its output.'''
+        return subprocess.check_output(("docker", "exec", self.name) + command)
+
+
+class RadioDocker(Radio):
+    '''Docker implementation of a radio.'''
+    def __init__(self, agent: AlEntityDocker, iface_name: str):
+        self.iface_name = iface_name
+        ip_output = agent.command("ip", "-o",  "link", "list", "dev", self.iface_name).decode()
+        mac = re.search(r"link/ether (([0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})", ip_output).group(1)
+        super().__init__(agent, mac)
+
+
+def _get_bridge_interface(unique_id):
+    '''Use docker network inspect to get the docker bridge interface.'''
+    docker_network = 'prplMesh-net-{}'.format(unique_id)
+    docker_network_inspect_cmd = ('docker', 'network', 'inspect', docker_network)
+    inspect_result = subprocess.run(docker_network_inspect_cmd, stdout=subprocess.PIPE)
+    if inspect_result.returncode != 0:
+        # Assume network doesn't exist yet. Create it.
+        # This is normally done by test_gw_repeater.sh, but we need it earlier to be able to
+        # start tcpdump
+        # Raise an exception if it fails (check=True).
+        subprocess.run(('docker', 'network', 'create', docker_network), check=True,
+                       stdout=subprocess.DEVNULL)
+        # Inspect again, now raise if it fails (check=True).
+        inspect_result = subprocess.run(docker_network_inspect_cmd, check=True,
+                                        stdout=subprocess.PIPE)
+
+    inspect = json.loads(inspect_result.stdout)
```

Comment:
Once again, sanity checks might be useful here (e.g. check that `inspect is not None`)

Reply:
json.loads will raise an exception if it's not JSON, and the lookups will raise KeyError. So I don't see why we should bother.